### ISSUE TITLE
Accept plugins "bundled" as tid/json in the plugin directory

### ISSUE THREAD

**Issue by felixhayashi** [TiddlyWiki/TiddlyWiki5#2230]
It would be great if a user could drop a `.tid` file in the plugins directory of tiddlywiki and they were recognized as plugins if they are of type plugin, i.e. more precisely they have a field `"plugin-type": "plugin"`. Also in this case, the `plugin.info` file should not be required. A bundled plugin already includes all that is needed since the `plugin.info` is already merged into the bundle.

The benefit? A single file that works everywhere: I can drag n drop it into the browser and I can put it into the plugin dir using nodejs. Very convenient.

**edit** I just tried putting a packed plugin (tiddlymap.json) into the plugin dir of tw along with the plugin.info, which failed: tw's `$tw.loadPluginFolder` interpreted this file as a file of the plugin and bundled the plugin as plugin again. It would be great if TW would acknowledge bundled.

**Comment by tobibeer** [comment]
you mean from a subfolder of the plugins directory right? actually, I think it should be a username folder inside as usual and then again a folder for each 'bundle'

tangentially related, I'd also appreciate if there was a way to create shadows on the fly… as we could in tw2 …to display dynamically created stuff in dynamically created tiddlers without having any of them persisted

I only mention this, because both are quite related to how plugins behave these days and the constructs they leverage compared to how that worked (differently) in TW2… providing some extra capability in TW2 that I find missing in TW5

**Comment by felixhayashi** [comment]
> you mean from a subfolder of the plugins directory right?

Yes, of course, sorry for not mentioning it. First the `username` then the plugin name as it is currently the case. Actually a separation by version would also make sense, but this is another topic…

**Comment by pmario** [comment]
Drag and drop importing a plugin into a client server configuration does exactly that.
See: Option 4 in [Installing custom plugins on Node.js](https://tiddlywiki.com/#Installing%20custom%20plugins%20on%20Node.js:%5B%5BInstalling%20custom%20plugins%20on%20Node.js%5D%5D%20%5B%5BManually%20installing%20a%20plugin%5D%5D)
Option 4: [Manually installing a plugin](https://tiddlywiki.com/#Manually%20installing%20a%20plugin)
